### ISSUE TITLE
org.eclipse.che.maven.server.MavenServerMain does not respect container env vars

### ISSUE THREAD

**Issue by jerboaa** [openshiftio/openshift.io#1411]
For Che workspaces the process with main class `org.eclipse.che.maven.server.MavenServerMain` has no way of getting tuned to the specific container use case. That is, no matter whether it'll run on a physical machine or a Docker container with a set memory limit it will always set things up according to the physical machine (i.e. a node on OpenShift) instead of the container limit.

Steps to reproduce:
1. Open any che workspace
2. Open a remote shell to the workspace pod
3. `$ set $(ps ax | grep MavenServerMain | grep -v grep)`
4. `$ jcmd $1 VM.flags`
5. `$ free -b`

Step 4 and 5 above produce:

    $ jcmd $1 VM.flags
    397:
    -XX:CICompilerCount=4 -XX:InitialHeapSize=526385152 -XX:MaxHeapSize=536870912 
    -XX:MaxNewSize=178782208 -XX:MinHeapDeltaBytes=524288 -XX:NewSize=175112192 
    -XX:OldSize=351272960 -XX:+UseCompressedClassPointers -XX:+UseCompressedOops 
    -XX:+UseParallelGC 
    $ free -b
              total        used        free      shared  buff/cache   available
     Mem:    33567498240  4348420096 22876745728  1638952960  6342332416 26970857472
     Swap:             0           0           0

Given that the default JVM ergonomics are `1/64` for `-XX:InitialHeapSize`, this matches the ***physical*** host. `33567498240/64 ~= 526385152`. This is wrong. It should really use the container limit. If the forked process would respect some environment variable, say `MAVEN_OPTS`, to override default GC/JVM flags this could get fixed. But looking at https://github.com/eclipse/che/blob/master/plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/MavenServerManager.java#L196 the list of JVM parameters is hard-coded.

The net effect of this is that the process will either die or get OOM killed in certain situations. It appears to be pure luck that it works for the current Che setup on OSIO.

The JVM supports a couple of options to override the actual physical memory it should be using. E.g. `MaxRAM` or `UseCGroupMemoryLimitForHeap`, but neither can be set as it's hard-coded.

**Comment by jerboaa** [comment]
@ibuziuk @l0rd FYI: this seems to inhibit memory tuning of che workspaces on OSIO.

**Comment by skabashnyuk** [comment]
Could be fixed by https://github.com/openshiftio/openshift.io/issues/1411

**Comment by joshuawilson** [comment]
This is #1411, an issue can't fix itself. ;-)
You have a link to the PR?

**Comment by skabashnyuk** [comment]
@joshuawilson Sorry. Correct link https://github.com/eclipse/che/issues/7726

**Comment by joshuawilson** [comment]
Fixed would be the wrong word here, fixed/fixes are keywords in github that link to the PR that "fixes" it. I was about to ask if it was in prod then clicked it to realize it was another issue.

This is more of "being worked from a different issue" kind of link.

**Comment by skabashnyuk** [comment]
Yeah. Unfortunately, it's only a registered issue for now. Hope I have better news next time.

**Comment by ibuziuk** [comment]
@l0rd do you happen to know if @vrubezhny is working on verification of this issue on osio ?

**Comment by skabashnyuk** [comment]
@ibuziuk FYI it would be much easier to do with che6 in comparison to che5

**Comment by l0rd** [comment]
@ibuziuk yes @vrubezhny should be working on this issue. 

@skabashnyuk yes actually @vrubezhny should verify that with Che6 this is fixed.
